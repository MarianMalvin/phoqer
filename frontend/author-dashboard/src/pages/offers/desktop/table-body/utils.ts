import { TableCellSizeEnum } from 'phoqer';

export interface HeaderType {
    key: string;
    title: string;
}

export const headerOrder: HeaderType[] = [
    { key: 'status', title: 'Status' },
    { key: 'options', title: 'Options' },
    { key: 'images', title: 'Images' },
    { key: 'title', title: 'Title' },
    { key: 'category', title: 'Category' },
    { key: 'reviews', title: 'Reviews' },
    { key: 'price', title: 'Price' },
    { key: 'sale_percentage', title: 'Sale percentage' },
    { key: 'sale_description', title: 'Sale description' },
    { key: 'description', title: 'Description' },
];

export const sizeMap: { [key: string]: TableCellSizeEnum } = {
    status: TableCellSizeEnum.Small,
    options: TableCellSizeEnum.Middle,
    images: TableCellSizeEnum.Small2Extra,
    title: TableCellSizeEnum.Large5Extra,
    description: TableCellSizeEnum.Large8Extra,
    sale_percentage: TableCellSizeEnum.MiddleExtra,
    sale_description: TableCellSizeEnum.Large4Extra,
    reviews: TableCellSizeEnum.Middle,
    price: TableCellSizeEnum.Middle,
    category: TableCellSizeEnum.Middle,
};
