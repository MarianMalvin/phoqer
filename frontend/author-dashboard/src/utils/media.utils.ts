export const isTablet = window.innerWidth > 768;
