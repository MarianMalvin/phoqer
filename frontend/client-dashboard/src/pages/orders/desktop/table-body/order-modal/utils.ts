export const cellMap = [
    { key: 'title', title: 'Title' },
    { key: 'status', title: 'Status' },
    { key: 'date', title: 'Created' },
    { key: 'expired', title: 'Expired' },
    { key: 'price', title: 'Price' },
    { key: 'category', title: 'Category' },
    { key: 'country', title: 'Country' },
    { key: 'city', title: 'City' },
    { key: 'zip', title: 'ZIP' },
    { key: 'address', title: 'Address' },
    { key: 'comment', title: 'Comment' },
    { key: 'sale', title: 'Sale' },
    { key: 'sale_description', title: 'Sale Description' },
];
