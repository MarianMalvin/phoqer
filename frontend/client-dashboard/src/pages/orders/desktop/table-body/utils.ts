import { TableCellSizeEnum } from 'phoqer';

export interface HeaderType {
    key: string;
    title: string;
}

export const headerOrder: HeaderType[] = [
    { key: 'status', title: 'Status' },
    { key: 'options', title: 'Options' },
    { key: 'images', title: 'Images' },
    { key: 'title', title: 'Title' },
    { key: 'price', title: 'Price' },
    { key: 'user', title: 'Author' },
    { key: 'address', title: 'Address' },
    { key: 'comment', title: 'Comment' },
    { key: 'date', title: 'Created' },
    { key: 'expired', title: 'Expiration Date' },
];

export const sizeMap: { [key: string]: TableCellSizeEnum } = {
    status: TableCellSizeEnum.Small,
    options: TableCellSizeEnum.Middle,
    title: TableCellSizeEnum.Large3Extra,
    images: TableCellSizeEnum.Small2Extra,
    price: TableCellSizeEnum.Middle,
    address: TableCellSizeEnum.Large4Extra,
    comment: TableCellSizeEnum.Large3Extra,
    user: TableCellSizeEnum.LargeExtra,
    date: TableCellSizeEnum.Large2Extra,
    expired: TableCellSizeEnum.Large2Extra,
};
