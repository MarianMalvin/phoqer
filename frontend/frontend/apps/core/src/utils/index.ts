export * from './recentOffers';
export * from './errorHandler';
