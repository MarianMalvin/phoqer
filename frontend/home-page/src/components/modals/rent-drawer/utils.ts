import { RentPayload } from './type';

export const RENT_DRAWER_DATA_KEY = 'rent-drawer-data';

export const initialValues: RentPayload = {
    country: '',
    city: '',
    zip: '',
    address: '',
    comment: '',
};

export const getInitialValues = (): RentPayload => {
    try {
        const json = localStorage.getItem(RENT_DRAWER_DATA_KEY);
        if (!json) {
            return initialValues;
        }

        const data = JSON.parse(json);
        return data || initialValues;
    } catch {
        return initialValues;
    }
};

export const setInitialValues = (value: RentPayload): void => {
    try {
        localStorage.setItem(RENT_DRAWER_DATA_KEY, JSON.stringify(value));
    } catch (error) {
        console.log(error);
    }
};
