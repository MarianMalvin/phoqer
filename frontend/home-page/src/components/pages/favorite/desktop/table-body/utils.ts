export interface HeaderType {
    key: string;
    title: string;
}

export const headerOrder: HeaderType[] = [
    { key: 'image', title: 'Image' },
    { key: 'options', title: 'Rent now' },
    { key: 'title', title: 'Title' },
    { key: 'price', title: 'Price' },
    { key: 'sale', title: 'Sale' },
    { key: 'category', title: 'Category' },
];

export const sizeMap: { [key: string]: number } = {
    image: 10,
    title: 45,
    price: 15,
    sale: 10,
    category: 12,
    options: 18,
};
