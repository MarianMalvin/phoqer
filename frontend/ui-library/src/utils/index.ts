export * from './copy';
export * from './css-property';
export * from './get-css-class';
export * from './label-size-map';
