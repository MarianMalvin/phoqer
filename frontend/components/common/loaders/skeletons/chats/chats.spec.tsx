import { render } from '../../../../../tests/tests.utils';
import ChatsLoader from './index';

describe('ChatsLoader component', () => {
    it('matches snapshot, without props', () => {
        const { asFragment } = render(<ChatsLoader />, {});
        expect(asFragment()).toMatchSnapshot();
    });

    it('matches snapshot, with amount', () => {
        const { asFragment } = render(<ChatsLoader amount={5} />, {});
        expect(asFragment()).toMatchSnapshot();
    });

    it('matches snapshot, with className', () => {
        const { asFragment } = render(<ChatsLoader amount={5} className="test" />, {});
        expect(asFragment()).toMatchSnapshot();
    });
});
