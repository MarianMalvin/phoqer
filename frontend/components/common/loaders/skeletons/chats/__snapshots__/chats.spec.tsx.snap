// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatsLoader component matches snapshot, with amount 1`] = `
<DocumentFragment>
  <ul>
    <li
      class="item-0-2-12"
    >
      <div
        class="wrp-0-2-16 img-0-2-13"
      >
        <div
          class="item-0-2-17"
        />
      </div>
      <div
        class="text-0-2-14"
      >
        <div
          class="text-0-2-19"
        />
        <div
          class="text-0-2-19"
        />
        <div
          class="text-0-2-19 short-0-2-20"
        />
      </div>
    </li>
    <li
      class="item-0-2-12"
    >
      <div
        class="wrp-0-2-16 img-0-2-13"
      >
        <div
          class="item-0-2-17"
        />
      </div>
      <div
        class="text-0-2-14"
      >
        <div
          class="text-0-2-19"
        />
        <div
          class="text-0-2-19"
        />
        <div
          class="text-0-2-19 short-0-2-20"
        />
      </div>
    </li>
    <li
      class="item-0-2-12"
    >
      <div
        class="wrp-0-2-16 img-0-2-13"
      >
        <div
          class="item-0-2-17"
        />
      </div>
      <div
        class="text-0-2-14"
      >
        <div
          class="text-0-2-19"
        />
        <div
          class="text-0-2-19"
        />
        <div
          class="text-0-2-19 short-0-2-20"
        />
      </div>
    </li>
    <li
      class="item-0-2-12"
    >
      <div
        class="wrp-0-2-16 img-0-2-13"
      >
        <div
          class="item-0-2-17"
        />
      </div>
      <div
        class="text-0-2-14"
      >
        <div
          class="text-0-2-19"
        />
        <div
          class="text-0-2-19"
        />
        <div
          class="text-0-2-19 short-0-2-20"
        />
      </div>
    </li>
    <li
      class="item-0-2-12"
    >
      <div
        class="wrp-0-2-16 img-0-2-13"
      >
        <div
          class="item-0-2-17"
        />
      </div>
      <div
        class="text-0-2-14"
      >
        <div
          class="text-0-2-19"
        />
        <div
          class="text-0-2-19"
        />
        <div
          class="text-0-2-19 short-0-2-20"
        />
      </div>
    </li>
  </ul>
</DocumentFragment>
`;

exports[`ChatsLoader component matches snapshot, with className 1`] = `
<DocumentFragment>
  <ul
    class="test"
  >
    <li
      class="item-0-2-22"
    >
      <div
        class="wrp-0-2-26 img-0-2-23"
      >
        <div
          class="item-0-2-27"
        />
      </div>
      <div
        class="text-0-2-24"
      >
        <div
          class="text-0-2-29"
        />
        <div
          class="text-0-2-29"
        />
        <div
          class="text-0-2-29 short-0-2-30"
        />
      </div>
    </li>
    <li
      class="item-0-2-22"
    >
      <div
        class="wrp-0-2-26 img-0-2-23"
      >
        <div
          class="item-0-2-27"
        />
      </div>
      <div
        class="text-0-2-24"
      >
        <div
          class="text-0-2-29"
        />
        <div
          class="text-0-2-29"
        />
        <div
          class="text-0-2-29 short-0-2-30"
        />
      </div>
    </li>
    <li
      class="item-0-2-22"
    >
      <div
        class="wrp-0-2-26 img-0-2-23"
      >
        <div
          class="item-0-2-27"
        />
      </div>
      <div
        class="text-0-2-24"
      >
        <div
          class="text-0-2-29"
        />
        <div
          class="text-0-2-29"
        />
        <div
          class="text-0-2-29 short-0-2-30"
        />
      </div>
    </li>
    <li
      class="item-0-2-22"
    >
      <div
        class="wrp-0-2-26 img-0-2-23"
      >
        <div
          class="item-0-2-27"
        />
      </div>
      <div
        class="text-0-2-24"
      >
        <div
          class="text-0-2-29"
        />
        <div
          class="text-0-2-29"
        />
        <div
          class="text-0-2-29 short-0-2-30"
        />
      </div>
    </li>
    <li
      class="item-0-2-22"
    >
      <div
        class="wrp-0-2-26 img-0-2-23"
      >
        <div
          class="item-0-2-27"
        />
      </div>
      <div
        class="text-0-2-24"
      >
        <div
          class="text-0-2-29"
        />
        <div
          class="text-0-2-29"
        />
        <div
          class="text-0-2-29 short-0-2-30"
        />
      </div>
    </li>
  </ul>
</DocumentFragment>
`;

exports[`ChatsLoader component matches snapshot, without props 1`] = `
<DocumentFragment>
  <ul>
    <li
      class="item-0-2-2"
    >
      <div
        class="wrp-0-2-6 img-0-2-3"
      >
        <div
          class="item-0-2-7"
        />
      </div>
      <div
        class="text-0-2-4"
      >
        <div
          class="text-0-2-9"
        />
        <div
          class="text-0-2-9"
        />
        <div
          class="text-0-2-9 short-0-2-10"
        />
      </div>
    </li>
  </ul>
</DocumentFragment>
`;
